app/components/AccountSetting/AccountSettingSection.js

```diff
@@ -6,9 +6,21 @@ import AccountSettingAction from './AccountSettingAction';
 import Header from 'components/Header/Header';
 
 export default class AccountSettingSection extends Component {
+
   constructor(props) {
     super(props);
     this.navigateToPage = this.navigateToPage.bind(this);
+    this.handleEmailChange = this.handleEmailChange.bind(this);
+    this.state = { email: this.props.email};
+    this.key = '';
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { location } = nextProps;
+    if (location.query.clear == 'true' && this.key != location.key) {
```

Comment:
I need somehow to this.setState({ email: this.props.email}) in AccountSettingSection.js only once, otherwise component is rerendered infinitly. If there is any better way, please let me know